NewRelic/AdaptiveInteractor.php

```diff
@@ -139,4 +139,29 @@ public function setUserAttributes(string $userValue, string $accountValue, strin
     {
         return $this->interactor->setUserAttributes($userValue, $accountValue, $productValue);
     }
+
+    public function getTraceMetadata(): array
+    {
+        return $this->interactor->getTraceMetadata();
```

Comment:
because of BC (see my previous comment) you don't know if the injected service implement this method. You should add a check

```php
if (!method_exists($this->interaction, 'getTraceMetadata')) {
  throw new BadMethodException('The decorated interaction does not implement this method');
}
```

Reply:
It's the same code in the same library. Why all the effort?
Isn't this making everything super complicated? :)

Reply:
We can also tag version 3.0?

Reply:
Changed the version in the change log to 3.0.0 :) this should keep the code nice and clean. 

Good feedback on the BC break, I didn't realize that others extend the interface to build their own implementations.

Reply:
Ah, i just did it ;) let me know what you think.